examples/medplum-demo-bots/src/deduplication/merge-bot.ts

```diff
@@ -0,0 +1,108 @@
+import { BotEvent, createReference, deepClone, getQuestionnaireAnswers, MedplumClient } from '@medplum/core';
+import { Patient, QuestionnaireResponse, Reference, Resource } from '@medplum/fhirtypes';
+
+interface MergedPatients {
+  readonly src: Patient;
+  readonly target: Patient;
+}
+
+interface Subject {
+  subject?: Reference<Patient>;
+}
+
+type ResourceWithSubject = Resource & Subject;
+
+export function linkPatientRecords(src: Patient, target: Patient): MergedPatients {
+  const targetLinks = target.link ?? [];
+  targetLinks.push({ other: createReference(src), type: 'replaces' });
+  const targetCopy = deepClone(target);
+
+  const srcLinks = src.link ?? [];
+  srcLinks.push({ other: createReference(target), type: 'replaced-by' });
+  const srcCopy = deepClone(src);
+  return { src: { ...srcCopy, link: srcLinks, active: false }, target: { ...targetCopy, link: targetLinks } };
+}
+
+export function mergeContactInfo(src: Patient, target: Patient, fields?: Partial<Patient>): MergedPatients {
+  const srcIdentifiers = src.identifier ?? [];
+  const mergedIdentifiers = srcIdentifiers.map((identifier) => ({
+    ...identifier,
+    use: 'old' as 'usual' | 'official' | 'temp' | 'secondary' | 'old',
```

Comment:
I see what you did here, but you can also use `as Identifier['use']` for the cast